tests/util/pilotServer.go

```diff
@@ -37,10 +42,59 @@ var (
 
 	// MockPilotGrpcAddr is the address to be used for grpc connections.
 	MockPilotGrpcAddr string
-	fsRoot            string
-	stop              chan struct{}
+
+	// MockPilotHttpPort is the dynamic port for pilot http
+	MockPilotHTTPPort int
+
+	// MockPilotGrpcPort is the dynamic port for pilot grpc
+	MockPilotGrpcPort int
+
+	fsRoot string
+	stop   chan struct{}
+)
+
+var (
+	// IstioTop has the top of the istio tree, matches the env variable from make.
+	IstioTop = os.Getenv("TOP")
+
+	// IstioSrc is the location if istio source ($TOP/src/istio.io/istio
+	IstioSrc = os.Getenv("ISTIO_GO")
+
+	// IstioBin is the location of the binary output directory
+	IstioBin = os.Getenv("ISTIO_BIN")
+
+	// IstioOut is the location of the output directory ($TOP/out)
+	IstioOut = os.Getenv("ISTIO_OUT")
```

Comment:
are these all existing env variables? if not, consider using flags instead when possible.

Reply:
They are existing env used in make and scripts, and we are actually trying to use env variables in some cases instead of flags ( like KUBECONFIG, the various fine tuning variables, paths).